examples/multi-modal-data-movielens/02-Data-Enrichment.ipynb

```diff
@@ -0,0 +1,472 @@
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>movieId</th>\n",
+       "      <th>imdbId</th>\n",
+       "      <th>tmdbId</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>1</td>\n",
+       "      <td>114709</td>\n",
+       "      <td>862.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>2</td>\n",
+       "      <td>113497</td>\n",
+       "      <td>8844.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>3</td>\n",
+       "      <td>113228</td>\n",
+       "      <td>15602.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>4</td>\n",
+       "      <td>114885</td>\n",
+       "      <td>31357.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>5</td>\n",
+       "      <td>113041</td>\n",
+       "      <td>11862.0</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   movieId  imdbId   tmdbId\n",
+       "0        1  114709    862.0\n",
+       "1        2  113497   8844.0\n",
+       "2        3  113228  15602.0\n",
+       "3        4  114885  31357.0\n",
+       "4        5  113041  11862.0"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "links.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "62423"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "links.imdbId.nunique()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from tqdm import tqdm\n",
+    "import pickle"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "movies_infos = {}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "for cnt, row in tqdm(links.iterrows(), total=len(links)):\n",
```

Comment:
The process of fetching the metadata is estimated to take ~30 hours on my machine. I see that the IMDB dataset is available for download from S3 and that imdbpy can fetch data from a local database instead with a little bit of client configuration. Seems like it would be faster to download the data in bulk and ingest it into something like a SQLite DB with the scripts provided by IMDbPY.

Reply:
yep if this shortcut works, should be great.

Reply:
I'm using 8 parallel workers and it seems the time is cut quite substantially, now ~3.5h. 
Using more workers initially (~128) and was banned by their anti DoS ;)

Reply:
5 parallel threads seem a sweet spot. Taking ~6h.